JetUtilities/src/L2Creator.cc

```diff
@@ -186,200 +191,302 @@ void L2Creator::loopOverAlgorithms() {
+      ss<<hl_rsp.minimum(0,ieta)<<"to"<<hl_rsp.maximum(0,ieta);
+      vabsrsp_eta.back()->SetName(("AbsRspVsRefPt_JetEta"+ss.str()).c_str());
+      vabscor_eta.back()->SetName(("AbsCorVsJetPt_JetEta"+ss.str()).c_str());
+    }
+
 
     //
     // only add points to the graphs if the current histo is not empty
     // the current setting might be a little high
     //
-    if (hrsp->GetEntries() > 10) {//hrsp->Integral()!=0) { 
-
-			TF1*  frsp    = (TF1*)hrsp->GetListOfFunctions()->Last();
-      //std::cout << "hrspName = " << hrsp->GetName() << ": frsp = " << frsp << std::endl;
-			TH1F* hrefpt  = hl_refpt.object(indices);
-			TH1F* hjetpt  = hl_jetpt.object(indices);
-
-			assert(hrefpt->GetEntries()>0&&hjetpt->GetEntries()>0);
-
-			double refpt  =hrefpt->GetMean();
-			double erefpt =hrefpt->GetMeanError();
-			double jetpt  =hjetpt->GetMean();
-			double ejetpt =hjetpt->GetMeanError();
-
-			double peak;
-			double epeak;
-			if(alg.find("calo")!=string::npos) {
-				peak = (frsp==0 || !mpv)?hrsp->GetMean():frsp->GetParameter(1);
-				epeak = (frsp==0 || !mpv)?hrsp->GetMeanError():frsp->GetParError(1);
-			}
-			else if(alg.find("pf")!=string::npos) {
-				peak = (frsp==0 || !mpv)?hrsp->GetMean():frsp->GetParameter(1);
-				epeak = (frsp==0 || !mpv)?hrsp->GetMeanError():frsp->GetParError(1);
-			}
-			else {
-				peak = (frsp==0 || !mpv)?hrsp->GetMean():frsp->GetParameter(1);
-				epeak = (frsp==0 || !mpv)?hrsp->GetMeanError():frsp->GetParError(1);
-			}
-
-			double absrsp = peak;
-			double eabsrsp = epeak;
-			double abscor = 0.0;
-			double eabscor = 0.0;
-
-			if (absrsp > 0) {  
-				abscor  =1.0/absrsp;
-				eabscor = abscor*abscor*epeak;
-			} 
-			if ((abscor>0) && (absrsp>0) && (eabscor>1e-5) && (eabscor/abscor<0.5) && (eabsrsp>1e-4) && (eabsrsp/absrsp<0.5)) { 
-				int n = vabsrsp_eta.back()->GetN();
-				vabsrsp_eta.back()->SetPoint     (n,refpt, absrsp);
-				vabsrsp_eta.back()->SetPointError(n,erefpt,eabsrsp);
-				vabscor_eta.back()->SetPoint     (n,jetpt, abscor);
-				vabscor_eta.back()->SetPointError(n,ejetpt,eabscor);      
-			}
-			else cout << "absrsp " << absrsp << " and eabsrsp " << eabsrsp << " and abscor " << abscor << " and eabscor " << eabscor << endl;
-		}
+    if (hrsp->GetEntries()<1) continue;
+
+    TF1*  frsp {0};
+    frsp    = (TF1*)hrsp->GetListOfFunctions()->Last();
+    //std::cout << "hrspName = " << hrsp->GetName() << ": frsp = " << frsp << std::endl;
+    TH1F* hrefpt  = hl_refpt.object(indices);
+    TH1F* hjetpt  = hl_jetpt.object(indices);
+    cout<<"\nhrsp name: "<<hrsp->GetName()<<endl;
+    cout<<"graph name: "<<vabscor_eta.back()->GetName()<<endl;
+    cout<<"hjetptname: "<<hjetpt->GetName()<<endl;
+    cout<<"hrefptname: "<<hrefpt->GetName()<<endl;
+
+    assert(hrefpt->GetEntries()>0&&hjetpt->GetEntries()>0);
+
+    double refpt {hrefpt->GetMean()}; 
+    double erefpt{hrefpt->GetMeanError()};
+    double jetpt {hjetpt->GetMean()};
+    double ejetpt{hjetpt->GetMeanError()};
```

Comment:
No, this is fine. I just wanted to ask. Whenever there is a change I can't explain I like to understand the reasoning behind it.